packages/smart-wallet/src/smartWallet.js

```diff
@@ -399,21 +360,25 @@ export const prepareSmartWallet = (baggage, shared) => {
          * @returns {Promise<Amount>} amounts for deferred deposits will be empty
          */
         async receive(payment) {
-          const { brandPurses, paymentQueues: queues } = this.state;
+          const { paymentQueues: queues, bank, invitationPurse } = this.state;
+          const { registry, invitationBrand } = shared;
           const brand = await E(payment).getAllegedBrand();
 
           // When there is a purse deposit into it
-          if (brandPurses.has(brand)) {
-            const purse = brandPurses.get(brand);
+          if (registry.has(brand)) {
+            const purse = E(bank).getPurse(brand);
             // @ts-expect-error deposit does take a FarRef<Payment>
             return E(purse).deposit(payment);
+          } else if (invitationBrand === brand) {
+            // @ts-expect-error deposit does take a FarRef<Payment>
+            return E(invitationPurse).deposit(payment);
           }
 
           // When there is no purse, queue the payment
           if (queues.has(brand)) {
             queues.get(brand).push(payment);
           } else {
-            queues.init(brand, harden([payment]));
+            queues.init(brand, harden([payment])); // @@ this can't be right
```

Comment:
oh yeah we must not have coverage of this. You can't push onto a hardened array.

Good catch! Care to add the failing test 😬 ?

Reply:
8b6dd8418

Reply:
please replace `@@` with a link to the ticket